execution-engine/engine-tests/benches/transfer_bench.rs

```
@@ -87,6 +130,41 @@ fn transfer_to_account_multiple_deploys(builder: &mut LmdbWasmTestBuilder, accou
     builder.exec(exec_builder.build()).expect_success().commit();
 }
 
+/// Uses multiple exec requests with a single deploy to transfer tokens from purse to purse.
+/// Executes all transfers in batch determined by value of TRANSFER_BATCH_SIZE.
+fn transfer_to_purse_multiple_execs(builder: &mut LmdbWasmTestBuilder, purse_id: PurseId) {
+    let amount = U512::one();
+
+    // To see raw numbers take current time
+    for _ in 0..TRANSFER_BATCH_SIZE {
+        let exec_request = ExecuteRequestBuilder::standard(
+            TARGET_ADDR,
+            CONTRACT_TRANSFER_TO_PURSE,
+            (purse_id, amount),
+        )
+        .build();
+        builder.exec(exec_request).expect_success().commit();
+    }
+}
+
+/// Executes multiple deploys per single exec with based on TRANSFER_BATCH_SIZE.
+fn transfer_to_purse_multiple_deploys(builder: &mut LmdbWasmTestBuilder, purse_id: PurseId) {
+    let mut exec_builder = ExecuteRequestBuilder::new();
+
+    for i in 0..TRANSFER_BATCH_SIZE {
+        let deploy = DeployItemBuilder::default()
+            .with_address(TARGET_ADDR)
+            .with_payment_code(STANDARD_PAYMENT_CONTRACT, (U512::from(PER_RUN_FUNDING),))
+            .with_session_code(CONTRACT_TRANSFER_TO_PURSE, (purse_id, U512::one()))
+            .with_authorization_keys(&[PublicKey::new(TARGET_ADDR)])
+            .with_deploy_hash([2 + i as u8; 32]) // deploy_hash
```

If `TRANSFER_BATCH_SIZE` is big enough, couldn't this overflow and fail?

---

You're right, although for practical purposes `TRANSFER_BATCH_SIZE` is set to 3

---

True - it's likely not a problem, especially since this is test code and if we do change `TRANSFER_BATCH_SIZE` to 254 or above the test won't silently pass.  I was just thinking something like:
```rust
let deploy_hash = {
    let mut hash = [0; 32];
    (&mut hash[0..8]).copy_from_slice(&i.to_be_bytes());
    hash
};
```
or
```rust
let deploy_hash = {
    let mut hash = [0; 32];
    hash.chunks_mut(8).for_each(|chunk| chunk.copy_from_slice(&i.to_be_bytes()));
    hash
};
```
but that's maybe overkill.  Happy to approve as is.

---

@Fraser999 Ah, yes, fair point. I've done something like that thanks to your suggestion: https://github.com/CasperLabs/CasperLabs/pull/1301/commits/2c686ffce9ec2a4c76be65eb457bdf247da56450#diff-30e14e6a2151f30f682314b10f7e224cR37-R41